File: examples/fictional.py
from aerocalc import graph as g

A: g.NodeKey = "A"
B: g.NodeKey = "B"
C: g.NodeKey = "C"
D: g.NodeKey = "D"
E: g.NodeKey = "E"
F: g.NodeKey = "F"

NODES = {
    A: g.Node(A),
    B: g.Node(B),
    C: g.Node(C),
    D: g.Node(D),
    E: g.Node(E),
    F: g.Node(F),
}


EDGES = [
    g.Edge(A, B, set(), set()),
    g.Edge(A, C, set(), set()),
    g.Edge(A, D, set(), set()),
    g.Edge(A, E, set(), set()),
    g.Edge(B, D, set(), set()),
    g.Edge(B, E, set(), set()),
    g.Edge(C, D, set(), set()),
    g.Edge(C, E, set([B]), set()),
    g.Edge(D, F, set(), set()),
]

GRAPH = g.build_graph(NODES, EDGES)

NODES = set(A)

s = GRAPH.solve_all(NODES, A, F)
print(s)
assert s[0] == [A, D, F]

File: web/js/graphs/fictional.js
// Fictional example graph

function loadFictional() {
    const A = 'A', B = 'B', C = 'C', D = 'D', E = 'E', F = 'F';

    const nodes = new Map();
    [A, B, C, D, E, F].forEach(name => {
        nodes.set(name, new Node(name));
    });

    const edges = [
        new Edge(A, B, new Set(), new Set()),
        new Edge(A, C, new Set(), new Set()),
        new Edge(A, D, new Set(), new Set()),
        new Edge(A, E, new Set(), new Set()),
        new Edge(B, D, new Set(), new Set()),
        new Edge(B, E, new Set(), new Set()),
        new Edge(C, D, new Set(), new Set()),
        new Edge(C, E, new Set([B]), new Set()),
        new Edge(D, F, new Set(), new Set())
    ];

    const graph = new Graph(nodes, edges, new Set());

    return {
        name: 'Fictional',
        description: 'Simple pathfinding example',
        nodes: [A, B, C, D, E, F],
        edges: edges.map(e => ({ from: e.neighbor1, to: e.neighbor2 })),
        conditions: [],
        defaultKnown: [A],
        defaultConditions: [],
        graph: graph
    };
}
